src/subdomains/staking/application/services/staking-deposit.service.ts

```diff
@@ -65,14 +67,15 @@ export class StakingDepositService {
     await this.kycCheck.check(userId, walletId);
 
     const staking = await this.authorize.authorize(userId, stakingId);
+    if (staking.isBlocked) throw new BadRequestException('Staking is blocked');
```

Comment:
Is this a good idea? 🤔

Reply:
it is actually what we had before in addDeposit method inside staking, so nothing changes in principle, or am I missing something?

Reply:
Oh yes, you're right 👍  thought the check was if staking is active, which wouldn't make sense. But this is very good 👍